bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java

```diff
@@ -157,13 +158,11 @@ public void run() {
                         } catch (InterruptedException e) {
                         }
                     }
-                    logger.debug("Enabling channel item provider.");
-                    initialized = true;
-                    // simply call getAll() will create the items and notify all registered listeners automatically
-                    getAll();
-                    addRegistryChangeListeners();
-                }
-            });
+                    initialize();
+                });
```

Comment:
Shouldn't the `shutdown` method be called, so queued task are executed and the scheduler could be cleaned up after all tasks has been executed?